deploy/cluster_role.yaml

```diff
@@ -0,0 +1,100 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    app.kubernetes.io/component: controller
+    app.kubernetes.io/name: observatorium-operator
+    app.kubernetes.io/version: v0.1
+  name: observatorium-operator
```

Comment:
if we have some time, this would be amazing if we could implement in locutus that locutus generates the RBAC roles it needs as it knows the steps it might do upfront

Reply:
Hi @brancz, how do we plan to proceed with this?

Reply:
I think this is a nice-to-have that we can do later. It doesn't change the API surface of the operator or the external behavior, just the amount of manifests that we need to generate. This PR is sufficiently larger so let's get it merged :muscle: